src/go/MONIT/ggus_parser.go

```diff
@@ -291,25 +298,88 @@ func processResponse(url, format, accept, out string) {
 		}
 	}
 
+	return resp
+}
+
+func retrieveTicketDetails(ticketId int) (string, []TicketComment) {
+	url := fmt.Sprintf("https://ggus.eu/?mode=ticket_info&ticket_id=%d", ticketId)
+	resp := ggusRequest(url, "text/html")
+	fmt.Println(resp)
+
+	doc, _ := html.Parse(resp.Body)
+
+	description := ""
+	var comments []string
+
+	var f func(*html.Node, bool, bool)
+	f = func(n *html.Node, isDescription bool, isHistory bool) {
+		if n.Type == html.TextNode && strings.Trim(n.Data, " \n\t\r") != "" && isDescription {
+			description = description + strings.Trim(n.Data, " \n\t")
+		} else if n.Type == html.TextNode && strings.Trim(n.Data, " \n\t\r") != "" && isHistory {
+			comments = append(comments, strings.Trim(n.Data, " \n\t"))
+		}
+
+		for c := n.FirstChild; c != nil; c = c.NextSibling {
+			if c.Type == html.CommentNode && c.Data == "#######    Description   #######" {
```

Comment:
this is fragile, what if Description string will change? The better approach would be to define regexp patterns which it should match

Reply:
Hey, I agree that this is pretty fragile, but processing this HTML a bit tricky. 

To clarify on your suggestion, do you suggest a regext that matches for example "#+ + Description +#+"? Or something else?

Reply:
do not hardcode stuff, make it configurable, i.e. provide option to read regexp patterns to use and then code will read them, compile and use to match the html content. Doing this way code will allow to match ANY pattern since it will be configurable from command line.